powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/util/CRC16.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2018 Wultra s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.getlime.security.powerauth.crypto.lib.util;
+
+import java.nio.ByteBuffer;
+
+public class CRC16 {
```

Comment:
Just for note, there's ["Checksum"](https://docs.oracle.com/javase/7/docs/api/java/util/zip/Checksum.html) interface, perhaps it would be a good idea to implement it. 

This is not a blocker for accepting this PR, it's just an idea.

Reply:
Yes, makes sense, I will implement it same as in CRC32:
https://github.com/openjdk-mirror/jdk7u-jdk/blob/master/src/share/classes/java/util/zip/CRC32.java